Frontend/create season updates

Paalar opened the pull request:
- Lets admin update current season or create a new one
- Lets admin access a season through url or link with #271 
- Re-works  how application season works in the reducer
- Relies on backend from #256

PeterRydberg commented on frontend/src/components/CreateSeason/index.tsx:
```diff
@@ -168,11 +201,15 @@ class _CreateSeason extends Component<Props, IState> {
 const mapStateToProps = (state: IStore) => ({
+  currentSeason: state.applicationSeason.currentSeason,
+  id: state.applicationSeason.id,
   submitted: state.applicationSeason.submitted,
```
The fields above could be combined into applicationSeason and sent as one in props, but that might not be necessary

Paalar replied:
Agreed, but I preferred splitting them, but can combine them if others find it better as well.

PeterRydberg replied:
I think it's fine, I'll approve

PeterRydberg commented:
Fix merge conflict and type error (see CI log), then I'll approve

Paalar commented:
> Fix merge conflict and type error (see CI log), then I'll approve

Fixed merge conflicts.
The error has previously been fixed, but the CI doesn't understand. You can run all the commans (npm run lint, npm run test, flake8, python3 -m pytest) and they will all pass.

Adrianht commented:
When having already created a season, nothing happens if you click create new season with different dates

Adrianht commented:
When having already created a season, you can't update it, by setting new dates and clicking update season

Paalar commented:
> After creating a season, you can't go back to create season in the sidebar

Fixed.

> When having already created a season, nothing happens if you click create new season with different dates

This is a problem with how we do seasons on main page. Not part of this pr.


> When having already created a season, you can't update it, by setting new dates and clicking update season

Fixed.
